src/manifolds/SymmetricPositiveDefiniteAffineInvariant.jl

```diff
@@ -412,18 +412,39 @@ function parallel_transport_to!(M::SymmetricPositiveDefinite{N}, Y, p, X, q) whe
     return copyto!(Y, vtp)
 end
 
+@doc raw"""
+    riemannian_Hessian(M::SymmetricPositiveDefinite, p, G, H, X)
+
+The Riemannian Hessian can be computed as stated in Eq. (7.3) [Nguyen:2023](@cite).
+Let ``\nabla f(p)`` denote the Euclidean gradient `G`,
+``\nabla^2 f(p)[X]`` the Euclidean Hessian `H`, and
+``\operatorname{sym}(X) = \frac{1}{2}\bigl(X^{\mathrm{T}+X\bigr)```
+the symmetrization operator. Then the formula reads
+
+```math
+    \operatorname{Hess}f(p)[X]
+    =
+    p\operatorname{sym}(∇^2 f(p)[X])p
+    + \operatorname{sym}\Bigl( X\operatorname{sym}\bigl(∇ f(p)\bigr)p)
+```
+"""
+riemannian_Hessian(M::SymmetricPositiveDefinite, p, G, H, X)
+
+function riemannian_Hessian!(::SymmetricPositiveDefinite, Y, p, G, H, X)
+    # The following formulae even work for complex, H-symmetric positive definite matrices.
+    # the term within the seconds summands outer \sym:
```

Comment:
What does the second line of this comment mean?

Reply:
we can remove that it is basically what you ask for in the other terms, just that here I did that in place of `Y` to not allocate (so here I did see one could spare some allocations/multiplications by using Y). `the term` refers to what Y is in 436.

Reply:
I see. I can't figure out what does `"seconds summands outer \sym` means though, so maybe let's phrase it differently? Like "Y is used for an intermediate value in matrix symmetrization"?